pkg/agent/globalstate.go

```diff
@@ -480,7 +490,7 @@ func (s *lockedPodStatus) update(global *agentState, with func(podStatus) podSta
 		case podStatusExitPanicked:
 			newState = runnerMetricStatePanicked
 		}
-	} else if newStatus.monitorStuckAt(global.config).Before(now) {
+	} else if newStatus.isStuck(global, now) {
```

Comment:
I _think_ this should still be sound, but it makes `periodicallyRefreshState` a little weirder. I think ultimately it's probably the right call, but let's discuss - I'll book some time on your calendar.

Reply:
See this 20a66f353704170fae0f1aa760905f4ce3674a1e.

Reply:
And a bug fix: 51b2459d44f69418dde70800c615a7f0d6419b52

Reply:
> bug fix

hah, I think I've made that mistake before :)